dtfs-central-api/src/v1/controllers/portal/gef-deal/update-deal.controller.js

```diff
@@ -3,48 +3,51 @@ const { findOneDeal } = require('./get-gef-deal.controller');
 const db = require('../../../../drivers/db-client');
 
 const updateDeal = async (dealId, update) => {
-  if (ObjectId.isValid(dealId)) {
+  try {
+    if (!ObjectId.isValid(dealId)) {
+      return { status: 400, message: 'Invalid Deal Id' };
+    }
+
     const collection = await db.getCollection('deals');
     const originalDeal = await findOneDeal(dealId);
-
-    console.info('Updating Portal GEF deal.');
-
     const dealUpdate = {
       ...originalDeal,
       ...update,
       updatedAt: Date.now(),
     };
-
     const findAndUpdateResponse = await collection.findOneAndUpdate(
       { _id: { $eq: ObjectId(String(dealId)) } },
       { $set: dealUpdate },
       { returnNewDocument: true, returnDocument: 'after' }
     );
 
-    console.info('Updated Portal GEF deal');
-
     return findAndUpdateResponse.value;
+  } catch (error) {
+    console.error('Unable to update deal %s', dealId);
+    return { status: 400, message: error };
   }
-  return { status: 400, message: 'Invalid Deal Id' };
 };
 exports.updateDeal = updateDeal;
 
-// eslint-disable-next-line consistent-return
 exports.updateDealPut = async (req, res) => {
-  if (ObjectId.isValid(req.params.id)) {
-    const dealId = req.params.id;
+  try {
+    if (!ObjectId.isValid(req.params.id)) {
+      return res.status(400).send({ status: 400, message: 'Invalid Deal Id' });
+    }
 
+    const dealId = req.params.id;
     const { dealUpdate } = req.body;
 
-    await findOneDeal(dealId, async (existingDeal) => {
+    return findOneDeal(dealId, async (existingDeal) => {
```

Comment:
Same as above.

Reply:
In the screenshot you shared in your reply above, the error was caught because the error happened in `updateDeal` and `updateDeal` has its own `try-catch` which caught the error. It wasn't the `try-catch` in _this_ function that caught the error.

`findOneDeal` doesn't have its own `try-catch` and any errors it throws can't be caught by the `try-catch` you added to this function because we don't `await` `findOneDeal`. The `try-catch` in this function can only catch errors from the following lines:
```
if (!ObjectId.isValid(req.params.id)) {
      return res.status(400).send({ status: 400, message: 'Invalid Deal Id' });
    }

    const dealId = req.params.id;
    const { dealUpdate } = req.body;
```

Is that deliberate? Happy to have a call to chat about it :)